at_client/lib/src/transformer/request_transformer/put_request_transformer.dart

```diff
@@ -44,6 +48,12 @@ class PutRequestTransformer
       }
       updateVerbBuilder.dataSignature = await SignInPublicData.signInData(
           updateVerbBuilder.value, encryptionPrivateKey!);
+      // Encode the public data if it contains new line characters
+      if (updateVerbBuilder.value.contains('\n')) {
+        updateVerbBuilder.value =
+            AtEncoderImpl().encodeData(updateVerbBuilder.value, encodingType);
+        updateVerbBuilder.encoding = encodingType.toString().split('.')[1];
```

Comment:
please check how to get name from enum in a cleaner way. 
we can replace with encodingType.name using extensions
https://stackoverflow.com/questions/29567236/dart-how-to-get-the-name-of-an-enum-as-a-string

Reply:
This change is implemented.